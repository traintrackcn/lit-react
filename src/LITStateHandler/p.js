import LITPath from "../LITPath";

//for test purpose

export class RootPath extends LITPath{
    constructor(path){
        super(path);
        this.t = this.path.concat(['t']);
        this.a = this.path.concat(['a']);
    }
}



export default new RootPath(['r']);