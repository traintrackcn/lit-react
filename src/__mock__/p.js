import LITPath from "../LITPath";

//for test purpose

export class RootPath extends LITPath{
    constructor(path){
        super(path);
        this.property1 = this.path.concat(['property1']);
        this.guest = new GuestPath(this.path.concat(['guest']));
    }
}


class GuestPath extends LITPath {
    constructor(path){
        super(path);
        this.email = this.path.concat(['email']);
    }
}


export default new RootPath(['r']);