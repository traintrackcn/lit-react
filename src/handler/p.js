import LITPath from "../LITPath";

//for test purpose

class C extends LITPath{
    constructor(path){
        super(path);
        this.t = this.path.concat(['t']);
    }
}

export default new C(['r']);