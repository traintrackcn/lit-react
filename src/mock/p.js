import LITPath from "../LITPath";

//for test purpose

export class RootPath extends LITPath{
    constructor(path){
        super(path);
        this.property1 = this.path.concat(['property1']);
    }
}



export default new RootPath(['r']);