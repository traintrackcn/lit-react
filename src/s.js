import LITReducer from "./LITReducer";
import { createStore } from "redux";
import { combineReducers } from 'redux-immutable';
import LITStore from "./LITStore";

export const r = new LITReducer();
const rootReducer = combineReducers({
    r: r.reducer
});
const store = createStore(rootReducer);
export default new LITStore(store, r);