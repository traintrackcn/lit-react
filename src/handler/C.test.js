import StateU from './C';
import s from '../s';
import p from './p';
import { fromJS } from 'immutable';

const stateU = new StateU(p.t);
stateU.s = s;

describe('', () => {


    it('test set_collection', () => {

        stateU.del();

        const collection = [
            'a', 'b', 'c'
        ];
        stateU.set_collection(fromJS(collection));
        expect(stateU.get()).toMatchSnapshot();
        stateU.select(1);
        expect(stateU.get_value()).toMatchSnapshot();

        expect(stateU.get()).toMatchSnapshot();
    });

    it('test set_value', () => {

        stateU.del();

        const value = { 'key': 'value' }
        stateU.set_value(value);
        expect(stateU.get()).toMatchSnapshot();
        expect(stateU.get_value()).toMatchSnapshot();
    });


    it('test s ops', () => {
        s.set(p.a, 'a');
        expect(s.get(p.a)).toMatchSnapshot();
        s.set(p.a, undefined);
        expect(s.get(p.a)).toBeFalsy();
    });


});