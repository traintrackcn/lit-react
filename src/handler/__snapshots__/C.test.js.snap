// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test set_collection 1`] = `
Immutable.Map {
  "collection": Immutable.List [
    "a",
    "b",
    "c",
  ],
}
`;

exports[`test set_collection 2`] = `"b"`;

exports[`test set_collection 3`] = `
Immutable.Map {
  "collection": Immutable.List [
    "a",
    "b",
    "c",
  ],
  "index": 1,
  "value": "b",
}
`;

exports[`test set_value 1`] = `
Immutable.Map {
  "value": Immutable.Map {
    "key": "value",
  },
}
`;

exports[`test set_value 2`] = `
Immutable.Map {
  "key": "value",
}
`;
